external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -305,11 +277,12 @@ private[kafka010] class KafkaOffsetReader(
   def fetchLatestOffsets(
       knownOffsets: Option[PartitionOffsetMap]): PartitionOffsetMap =
     partitionsAssignedToConsumer { partitions => {
-      logDebug("Seeking to the end.")
-
       if (knownOffsets.isEmpty) {
-        consumer.seekToEnd(partitions)
-        partitions.asScala.map(p => p -> consumer.position(p)).toMap
+        val listOffsetsParams = partitions.asScala.map(_ -> OffsetSpec.latest()).toMap.asJava
```

Comment:
This can be defined earlier to deduplicate.

Reply:
Moved up.